### DISCUSSION TITLE
An Idea - what if we register all popups gobally?

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#5994] [Ideas]
Hello all,

this is just an idea for discussion and it's about popups and the `<$action-popup>` widget not knowing about the correct position

- In every button widget we could register a popup under `$tw.popup.registeredPopups` with the `{title: title, domNode: this.domNode, widget: this, wiki: this.wiki ... }` options
- Then, in the `<$action-popup>` widget in the `invokeAction` method we could first see if the popup with the given title is registered globally and if it is we trigger the popup in `$tw.popup.triggerRegisteredPopup` which has access to the domNode and its position and to the button widget that corresponds to it
- Like that the `<$action-popup>` widget can trigger popups at the correct positions

**Comment by pmario** [comment]
Not really 100% related. ... But the widget doesn't work for my setting at all. See: https://github.com/Jermolene/TiddlyWiki5/issues/5996

**Reply by BurningTreeC** [reply]
Hi @pmario - for my setting it works. It opens the "Tools" tab and if you go down to the "More" button you should see the open popup
